src/resources/dart-3-migration.md

```diff
@@ -0,0 +1,512 @@
+[`NullThrownError`]: https://api.dart.dev/dev/2.19.0-430.0.dev/dart-core/NullThrownError-class.html
+[`AbstractClassInstantiationError`]: https://api.dart.dev/stable/2.18.3/dart-core/AbstractClassInstantiationError-class.html
+[`CyclicInitializationError`]: https://api.dart.dev/dev/2.19.0-430.0.dev/dart-core/CyclicInitializationError-class.html
+[`BidirectionalIterator`]: https://api.dart.dev/dev/2.19.0-430.0.dev/dart-core/BidirectionalIterator-class.html
+
+#### `dart:async`
+
+- Removed the deprecated [`DeferredLibrary`][] class.
+  Use the [`deferred as`][] import syntax instead.
+
+[`DeferredLibrary`]: https://api.dart.dev/stable/2.18.4/dart-async/DeferredLibrary-class.html
+[`deferred as`]: https://dart.dev/guides/language/language-tour#deferred-loading
+
+#### `dart:developer`
+
+- Removed the deprecated [`MAX_USER_TAGS`][] constant.
+  Use [`maxUserTags`][] instead.
+
+- Removed the deprecated [`Metrics`][], [`Metric`][], [`Counter`][],
+  and [`Gauge`][] classes as they have been broken since Dart 2.0.
+
+[`MAX_USER_TAGS`]: https://api.dart.dev/stable/dart-developer/UserTag/MAX_USER_TAGS-constant.html
+[`maxUserTags`]: https://api.dart.dev/beta/2.19.0-255.2.beta/dart-developer/UserTag/maxUserTags-constant.html
+[`Metrics`]: https://api.dart.dev/stable/2.18.2/dart-developer/Metrics-class.html
+[`Metric`]: https://api.dart.dev/stable/2.18.2/dart-developer/Metric-class.html
+[`Counter`]: https://api.dart.dev/stable/2.18.2/dart-developer/Counter-class.html
+[`Gauge`]: https://api.dart.dev/stable/2.18.2/dart-developer/Gauge-class.html
+
+#### `dart:html`
+
+- As previously announced, the deprecated `registerElement`
+  and `registerElement2` methods in `Document` and `HtmlDocument` have been
+  removed.  See [#49536](https://github.com/dart-lang/sdk/issues/49536) for
+  details.
+
+#### `dart:math`
+
+- The `Random` interface can only be implemented, not extended.
+
+#### `dart:io`
+
+- Update `NetworkProfiling` to accommodate new `String` ids
+  that are introduced in vm_service:11.0.0
+
+#### Symptom
+
+Dart analysis (e.g. in your IDE, or in `dart analyze`/`flutter analyze`) will fail with errors like:
+
+```
+error line 2 • Undefined class 'CyclicInitializationError'.
+```
+
+#### Migration
+
+Manually migrate away from using these APIs.
+
+### Extends & implements
+
+Dart 3 supports new class modifiers [*TODO* LINK] that can restrict the capabilities of a class.
+They have been applied to a number of classes in the core libraries.
+
+#### Scope
+
+This is a *versioned* change, that only applies to language version 3.0 or later.
+
+#### `dart:core`
+
+* The `Function` type can no longer be implemented, extended or mixed in. Since Dart 2.0, writing implements Function has been allowed for backwards compatibility, but it has not had any effect. In Dart 3.0, the Function type is final and cannot be subtyped, preventing code from mistakenly assuming it works.
```

Comment:
```suggestion
* The `Function` type can no longer be implemented, extended or mixed in.
Since Dart 2.0, writing `implements Function` has been allowed
for backwards compatibility, but it has not had any effect.
In Dart 3.0, the `Function` type is final and cannot be subtyped,
preventing code from mistakenly assuming it works.
```